processing/src/main/java/org/apache/druid/query/filter/vector/ArrayVectorValueMatcher.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.filter.vector;
+
+import com.google.common.base.Predicate;
+import org.apache.druid.query.filter.DruidPredicateFactory;
+import org.apache.druid.segment.column.ColumnType;
+import org.apache.druid.segment.column.TypeSignature;
+import org.apache.druid.segment.column.ValueType;
+import org.apache.druid.segment.vector.VectorObjectSelector;
+
+import javax.annotation.Nullable;
+
+public class ArrayVectorValueMatcher implements VectorValueMatcherFactory
+{
+  protected final TypeSignature<ValueType> columnType;
+  protected final VectorObjectSelector selector;
+
+  public ArrayVectorValueMatcher(
+      TypeSignature<ValueType> columnType,
+      VectorObjectSelector selector
+  )
+  {
+    this.columnType = columnType;
+    this.selector = selector;
+  }
+
+  @Override
+  public VectorValueMatcher makeMatcher(@Nullable String value)
+  {
+    throw new UnsupportedOperationException(
+        "Vectorized matcher cannot make string matcher for ARRAY types"
+    );
+  }
```

Comment:
this method is a specialization just for `SelectorFilter`

This could be called if you try to use a selector filter to match array columns, I'm working on wiring this up better so that the callers can throw a better validation exception so users know what they did wrong